src/object/srv_obj_migrate.c

```diff
@@ -847,6 +847,24 @@ migrate_fetch_update_single(struct migrate_one *mrone, daos_handle_t oh,
 		daos_iod_t	*iod = &mrone->mo_iods[i];
 		uint32_t	start_shard;
 
+		if (mrone->mo_iods[i].iod_size == 0) {
+			/* Since rebuild will be done within the stable
+			 * epoch, so no other IO should attach these
+			 * data during rebuild, but if the container is
+			 * being destroyed at the same time, then it may
+			 * not pull anything from remote target, so let's
+			 * avoid vos_obj_update() in this case, otherwise
+			 * it may cause VOS segfault.
+			 */
```

Comment:
The comment sounds like a general problem, but the code here is specific for EC. (see the DAOS_OC_IS_EC() check above).

And I don't see how do you differentiate a failed fetch vs. fetch on deleted container here. I think we do need a well considered solution to handle container destroy while rebuilding case.

Reply:
when the container is destroyed, basically it can not find any object, basically iod_size will be set to 0 anyway. So this is trying to avoid the case. 

And this can happen for EC object. or I am not sure I follow the question.